iXBRLViewerPlugin/__init__.py

```diff
@@ -269,10 +290,17 @@ def getLocalFile(self, file, relpath, request):
 
 
 def guiRun(cntlr, modelXbrl, attach, *args, **kwargs):
+    for generateOnCall in pluginClassMethods("iXBRLViewer.GenerateOnCall"):
```

Comment:
EdgarRenderer has a browser menu with multiple options for the launched viewer, such as redline behavior, etc.  I think in that configuration we don't want the regular plugin tools menu addition as well.

Reply:
If you enable EdgarRenderer there's no longer anyway to get the latest Arelle viewer js? That will come as a surprise for users that enable both the EdgarRenderer and Arelle viewers (there are many).

If we go forward with this path (I don't think we should) we would need a notice in the GUI and CLI that enabling the EdgarRenderer overrides the Arelle viewer and if you want the latest and greatest Arelle viewer you must disable the EdgarRenderer.

Reply:
View  (menu ideas)
   EdgarRenderer
       Show Filing Data (launch ixviewer-plus)
        other options
   Arelle Viewer
       Show Filing Data (launch arelle viewer)
           Edgar 23.4 version (1.14.09)
           Edgar preview version (1.14.10)
           Latest public version (1.99.99)

Reply:
> I think we can update the CLI hook to exit if `--save-viewer` isn't used (`saveViewerDest` isn't set).

I think that's already the case - if you don't specify `--save-viewer` it doesn't create a viewer (see 
https://github.com/Arelle/ixbrl-viewer/blob/master/iXBRLViewerPlugin/__init__.py#L146)

Reply:
It would need to be dynamic, as some GUI users (and possibly cmd line workflows) interweave Edgar submissions with ESEF work, and wouldn't want to have a lot of clicks and the cmd line workflow may have to sequence through different disclosureSystems without reloading or reconfiguring.

Reply:
That's fine.  If they want both EDGAR and standard iXBRL viewer functionality, they can load both plugins.